File: app/src/main/java/sabalan/paydar/mohtasham/ezibazi/api_service/user_actvity/UserRequestsService.java
package sabalan.paydar.mohtasham.ezibazi.api_service.user_actvity;

import android.content.Context;


import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

import sabalan.paydar.mohtasham.ezibazi.api_service.ApiRequest;
import sabalan.paydar.mohtasham.ezibazi.api_service.Urls;
import sabalan.paydar.mohtasham.ezibazi.model.RentRequest;
import sabalan.paydar.mohtasham.ezibazi.model.ShopRequest;

public class UserRequestsService {
  private Context context;
  public UserRequestsService(Context context){
    this.context = context;
  }


  public void getRentRequests( final onRentRequestsReceived onRentRequestsReceived){
    ApiRequest apiRequest = new ApiRequest(context);
    apiRequest.request(ApiRequest.GET, Urls.USER_RENT_REQUESTS, null, true, new ApiRequest.onDataReceived() {
      @Override
      public void onReceived(JSONObject response, int status, String message, boolean error) {
        if (error){
          onRentRequestsReceived.onReceived(status, message, new ArrayList<RentRequest>());
          return;
        }

        try {
          List<RentRequest> requests =  new ArrayList<>();
          JSONArray data = response.getJSONArray("data");
          for (int i=0 ; i<data.length() ; i++){
            JSONObject requestObj = data.getJSONObject(i);
            requests.add(RentRequest.Parser.parse(requestObj));
          }

          onRentRequestsReceived.onReceived(status, message, requests);
        } catch (JSONException e) {
          e.printStackTrace();
        }

      }
    });


  }



  public void getShopRequests( final onShopRequestsReceived onShopRequestsReceived){
    ApiRequest apiRequest = new ApiRequest(context);
    apiRequest.request(ApiRequest.GET, Urls.USER_BUY_REQUESTS, null, true, new ApiRequest.onDataReceived() {
      @Override
      public void onReceived(JSONObject response, int status, String message, boolean error) {
        if (error){
          onShopRequestsReceived.onReceived(status, message, new ArrayList<ShopRequest>());
          return;
        }

        try {
          List<ShopRequest> requests =  new ArrayList<>();
          JSONArray data = response.getJSONArray("data");
          for (int i=0 ; i<data.length() ; i++){
            JSONObject requestObj = data.getJSONObject(i);
            requests.add(ShopRequest.Parser.parse(requestObj));
          }

          onShopRequestsReceived.onReceived(status, message, requests);
        } catch (JSONException e) {
          e.printStackTrace();
        }
      }
    });
  }










  public interface onRentRequestsReceived{
    void onReceived(int status, String message, List<RentRequest> requests);
  }

  public interface onShopRequestsReceived{
    void onReceived(int status, String message, List<ShopRequest> requests);
  }



}

File: app/src/main/java/sabalan/paydar/mohtasham/ezibazi/api_service/user_actvity/UserRequestsService.kt
package sabalan.paydar.mohtasham.ezibazi.api_service.user_actvity

import android.content.Context


import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

import java.util.ArrayList

import sabalan.paydar.mohtasham.ezibazi.api_service.ApiRequest
import sabalan.paydar.mohtasham.ezibazi.api_service.Urls
import sabalan.paydar.mohtasham.ezibazi.model.RentRequest
import sabalan.paydar.mohtasham.ezibazi.model.ShopRequest

class UserRequestsService(private val context: Context) {


    fun getRentRequests(onRentRequestsReceived: onRentRequestsReceived) {
        val apiRequest = ApiRequest(context)
        apiRequest.request(ApiRequest.GET, Urls.USER_RENT_REQUESTS, null!!, true, object : ApiRequest.onDataReceived {
            override fun onReceived(response: JSONObject?, status: Int, message: String, error: Boolean) {
                if (error) {
                    onRentRequestsReceived.onReceived(status, message, ArrayList<RentRequest>())
                    return
                }

                try {
                    val requests = ArrayList<RentRequest>()
                    val data = response!!.getJSONArray("data")
                    for (i in 0 until data.length()) {
                        val requestObj = data.getJSONObject(i)
                        requests.add(RentRequest.Parser.parse(requestObj))
                    }

                    onRentRequestsReceived.onReceived(status, message, requests)
                } catch (e: JSONException) {
                    e.printStackTrace()
                }

            }
        })


    }


    fun getShopRequests(onShopRequestsReceived: onShopRequestsReceived) {
        val apiRequest = ApiRequest(context)
        apiRequest.request(ApiRequest.GET, Urls.USER_BUY_REQUESTS, null!!, true, object : ApiRequest.onDataReceived {
            override fun onReceived(response: JSONObject?, status: Int, message: String, error: Boolean) {
                if (error) {
                    onShopRequestsReceived.onReceived(status, message, ArrayList<ShopRequest>())
                    return
                }

                try {
                    val requests = ArrayList<ShopRequest>()
                    val data = response!!.getJSONArray("data")
                    for (i in 0 until data.length()) {
                        val requestObj = data.getJSONObject(i)
                        requests.add(ShopRequest.Parser.parse(requestObj))
                    }

                    onShopRequestsReceived.onReceived(status, message, requests)
                } catch (e: JSONException) {
                    e.printStackTrace()
                }

            }
        })
    }


    interface onRentRequestsReceived {
        fun onReceived(status: Int, message: String, requests: List<RentRequest>)
    }

    interface onShopRequestsReceived {
        fun onReceived(status: Int, message: String, requests: List<ShopRequest>)
    }


}
